tests/rptest/services/admin.py

```diff
@@ -226,34 +239,45 @@ def partition_transfer_leadership(self, namespace, topic, partition,
         path = f"partitions/{namespace}/{topic}/{partition}/transfer_leadership?target={target_id}"
         self._request("POST", path)
 
+    def get_partition_leader(self, *, namespace, topic, partition):
+        partition_info = self.get_partitions(topic=topic,
+                                             partition=partition,
+                                             namespace=namespace)
+
+        return partition_info['leader_id']
+
     def transfer_leadership_to(self, *, namespace, topic, partition, target):
         """
         Looks up current ntp leader and transfer leadership to target node, 
-        this operations is NOP when current leader is the same as target. 
+        this operations is NOP when current leader is the same as target.
+        If user pass None for target this function will choose random replica for new leader.
         If leadership transfer was performed this function return True
         """
-        target_id = self.redpanda.idx(target) if isinstance(
-            target, ClusterNode) else target
+        def has_leader():
+            return self.get_partition_leader(
+                namespace=namespace, topic=topic, partition=partition) != -1
 
         #  check which node is current leader
-
-        def _get_details():
-            p = self.get_partitions(topic=topic,
-                                    partition=partition,
-                                    namespace=namespace)
-            self.redpanda.logger.debug(
-                f"ntp {namespace}/{topic}/{partition} details: {p}")
-            return p
-
-        def _has_leader():
-            return _get_details()['leader_id'] != -1
-
-        wait_until(_has_leader,
+        wait_until(has_leader,
                    timeout_sec=30,
                    backoff_sec=2,
                    err_msg="Failed to establish current leader")
 
-        details = _get_details()
+        details = self.get_partitions(topic=topic,
+                                      partition=partition,
+                                      namespace=namespace)
+
+        self.redpanda.logger.debug(
+            f"ntp {namespace}/{topic}/{partition} details: {details}")
+
+        if target == None:
+            target_id = next(
+                filter(lambda r: r["node_id"] != details['leader_id'],
+                       details["replicas"]))["node_id"]
+        else:
+            target_id = self.redpanda.idx(target) if isinstance(
+                target, ClusterNode) else target
+
```

Comment:
the changes to `transfer_leadership_to` are in support of the commit `ducktape: Add test for transaction admin_api` and don't appear to be trivial. so these changes should be in a separate commit which explain the change being made.